### PR TITLE
cpf_segmentation

### PR DESCRIPTION
**Pull request by tojiboyevf**

### CODE REVIEW

File: cpf_segmentation/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update
+RUN apt-get install -y cmake pkg-config libeigen3-dev libopencv-dev libpcl-dev
+
+WORKDIR /app/src/src
+
+ADD ./src .
+
+WORKDIR /app/src/gco-v3.0
```
Comment (DmiitriyJarosh) [inline cpf_segmentation/Dockerfile]: Why do you change workdir every time? Is it possible to add not only to the current workdir?

File: cpf_segmentation/src/segmentation_test.cpp [hunk 2]
```diff
@@ -58,7 +58,8 @@ int main(int argc, char** argv){
 
 
   /// Get pcd path from command line
-  std::string pcd_filename = argv[1];
+  std::string pcd_filename ="input/";
```
Comment (DmiitriyJarosh) [inline cpf_segmentation/src/segmentation_test.cpp]: spaces (codestyle)

**Review by DmiitriyJarosh — CHANGES_REQUESTED** [review]

File: cpf_segmentation/Dockerfile (hunk 1 above)
Comment (tojiboyevf) [reply cpf_segmentation/Dockerfile]: Fixed

File: cpf_segmentation/src/segmentation_test.cpp (hunk 2 above)
Comment (DmiitriyJarosh) [reply cpf_segmentation/src/segmentation_test.cpp]: One more space is still skipped `std::string pcd_filename ="input/";` -> `std::string pcd_filename = "input/";`

File: cpf_segmentation/src/segmentation_test.cpp [hunk 3]
```diff
@@ -102,20 +100,16 @@ int main(int argc, char** argv){
...
-    // If no filename is given, get output filename from inputname (strip seperators and file extension)
-    if (outputname.empty () || (outputname.at (0) == '-'))
-    {
-      outputname = pcd_filename;
-      size_t dot = outputname.find_last_of ('.');
-      if (dot != std::string::npos)
-      outputname = outputname.substr (0, dot);
-    }
+    std::string outputname="output/";
+    outputname+=argv[1];
```
Comment (DmiitriyJarosh) [inline cpf_segmentation/src/segmentation_test.cpp]: Space everywhere `outputname+=argv[1];` -> `outputname += argv[1];`

**Review by DmiitriyJarosh — CHANGES_REQUESTED** [review]

**Review by DmiitriyJarosh — APPROVED** [review]

**Review by anastasiia-kornilova — APPROVED** [review]

**Pull request merged**